libsolidity/formal/SMTEncoder.cpp

```diff
@@ -39,7 +40,20 @@ bool SMTEncoder::visit(ContractDefinition const& _contract)
 	solAssert(m_currentContract == nullptr, "");
 	m_currentContract = &_contract;
 
-	initializeStateVariables(_contract);
+	createStateVariables(_contract);
+
+	// Look for all the constructor invocations bottom up.
+	for (auto const& base: _contract.annotation().linearizedBaseContracts)
+		if (auto const& constructor =  base->constructor())
+			for (auto const& invocation: constructor->modifiers())
+			{
+				auto refDecl = invocation->name()->annotation().referencedDeclaration;
+				if (auto const& baseContract = dynamic_cast<ContractDefinition const*>(refDecl))
+				{
+					solAssert(!m_baseConstructorCalls.count(baseContract), "");
```

Comment:
@chriseth This assumes that a base constructor can have at most one invocation from a derived contract. Is that true?

Reply:
This is similar to how these are collected in the code generator.